### ISSUE TITLE
Add DefaultCPUAffinity to systemd/system.conf

### ISSUE THREAD

**Issue by haasn** [systemd/systemd#9692]
The goal is to have all processes run with a given CPU affinity out of the box, so that I can take special `realtime` processes and isolate them on their own CPUs (to get rid of latency spikes under high system load). Doing this obviously requires making sure that nothing else can run on those CPUs.

Even though systemd.exec knows about `CPUAffinity`, it doesn't seem as though there's an equivalent `DefaultCPUAffinity=` section in `system.conf`. It was suggested I should file a bug report for this:

```
22:39 <killermoehre> haasn: you should file a feature request for this on github for being able to set a DefaultCPUAffinity
```

Incidentally, it doesn't seem as though there's any way via `systemd.resource-control` to control the `cpuset` CGroup controller? Another solution I considered is limiting `-.slice` to the non-realtime CPUs, but I don't see any way to do this. (Which I guess might be because apparently, there's no cgroupv2 replacement for the `cpuset` controller, but systemd only wants to support cgroupv2?)

**Comment by haasn** [comment]
Ah, it seems setting the similar `CPUAffinity=` option has the desired effect: All processes inheriting from `systemd` seem to inherit the CPU Affinity automatically. So I don't think any change needs to be made?

**Comment by poettering** [comment]
Yes, CPUAffinity= already does what you are asking for: it sets the affinity both for PID 1 and for all forked off processes. The latter may be overriden for individual units with CPUAffinity= in the unit files.

The "cpuset" controllers has very awkward semantics and it's future in cgroupsv2 is not clear, hence we don't support it right now. It's definitely our intention to support it eventually, as soon as cgroupsv2 supports it properly. That said, "cpuset" doesn't bring much benefit over CPUAffinity= besides the ability to change the affinity of a unit as a whole dynamically after it was created.

Anyway, I figure we should update the docs a bit and clarify that CPUAffinity= in systemd.conf applies both to the service manager itself and its forked off children. Hence labelling this documentation issue.

**Comment by poettering** [comment]
Doc fix waiting in #9694
